### PR TITLE
`Exam mode`: Fix an issue where the welcome page texts are not editable

### LINKED ISSUE #8970: `Exam mode`: Allow changing (or disabling) hard-coded welcome page texts
### Is your feature request related to a problem?

When conducting an exam using Artemis, the students are greeted with a welcome page.
This page contains some texts that are hard-coded in the [respective component](https://github.com/ls1intum/Artemis/blob/a6eade2a875cae1080a86344b5c294af894a4b7b/src/main/webapp/app/exam/participate/exam-start-information/exam-start-information.component.html#L20-L32):
https://github.com/ls1intum/Artemis/blob/a6eade2a875cae1080a86344b5c294af894a4b7b/src/main/webapp/i18n/en/exam.json#L152-L164

While some (parts) of these texts (e.g., `readCarefully`, `submissionPeriod`, `workingInstruction`) may be general enough, having all of these texts hard-coded seems tailored to a specific exam mode, making it impossible to use Artemis for exams with different modalities.

In particular, we plan to conduct an exam in August which will **not** be open-book, therefore the hard-coded texts are not fitting.
Some examples include:
- `workOnYourOwn`: `This is an open book exam (including using the Internet)`
This will not be the case for our exam
- `checkForPlagiarism`: `If you happen to copy code, for example, from a book or a website`
This will not be allowed in our exam
- `programmingSubmission`: `No actual tests will be run against your submission.`
This will also not be the case for us as we plan to provide some small sanity-check tests

### Describe the solution you'd like

Some possibilities I could think of are:
- Allow showing/hiding these texts in the exam settings
- Make these texts editable in the `Exam texts` section on the exam edit page
- Use these texts as a default-value for the `Exam start text`-field on the exam edit page

### Describe alternatives you've considered

The only alternative I could think of would be to tell students to ignore this part of the start page, though apart from being hard to understand (i.e., which parts are to be ignored and which contain the correct information) this may also lead to legal consequences due to contradictory statements.

##### The hard-coded texts at the exam start page:
Screenshot of the exam start screen (of a test run) containing the mentioned messages

### PR DESCRIPTION
**Pull request by edkaya**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

It fixes #8970. 

### Description

The texts in welcome page were hard-coded and therefore can not be changed by instructors. This implementation defines these texts as default start text and therefore can be edited by instructors. 

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate to Exams 
4. Click on Create Exam 
5. Scroll down and verify that you can see a prewritten start text 
6. Verify that you can change this start text 
7. Click Save 
8. Participate the exam as a student 
9. Verify that welcome page text is correct 

#### Exam Mode Testing

Prerequisites:
- 1 Exam
- 1 Student

1. Log in to Artemis
2. Go to the exam
3. Make sure that nothing changed in welcome page functionally

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2
#### Exam Mode Test
- [ ] Test 1
- [ ] Test 2

### Screenshots

Before 
Screenshot 2024-07-04 at 23 00 02

After
Screenshot 2024-07-04 at 23 00 12

Screenshot 2024-07-04 at 23 49 46

Screenshot 2024-07-04 at 23 49 28

## Summary by CodeRabbit

- **New Features**
  - Introduced default text for exam start, enhancing the exam setup process.
  - Added dynamic text updates for exam properties like maximum points, working time, and grace period in the exam update component.

- **Improvements**
  - Enhanced the exam participation cover by updating the structure and data bindings for better information display.
  - Shifted from static to dynamic text elements in the exam start information component for improved flexibility.

- **Localization**
  - Simplified exam instructions in both English and German by removing detailed guidelines related to exam conduct and resources.

### CODE REVIEW

**Review by sarpsahinalp — DISMISSED** [review]
code lgtm

File: src/main/webapp/app/exam/manage/exams/exam-update.component.ts [hunk 1]
```diff
@@ -446,6 +450,69 @@ export class ExamUpdateComponent implements OnInit, OnDestroy {
             dayjs(this.exam.exampleSolutionPublicationDate).isBefore(this.exam.endDate || null)
         );
     }
+    /**
+     * Default exam start text, which can be edited by instructors in the text editor
+     */
+    get examDefaultStartText(): string {
+        const warningForInstructionsText =
+            '<!-- PLEASE MAKE SURE TO READ ALL INSTRUCTIONS, CHECK EXAM INFORMATION CAREFULLY, AND ADAPT THEM TO YOUR SPECIFIC USE CASES AS NEEDED -->\n\n';
```
Comment (Strohgelaender) [inline src/main/webapp/app/exam/manage/exams/exam-update.component.ts:458]:
```suggestion
            '<!-- Please adapt this text for your exam -->\n\n';
```

**Review by Strohgelaender — APPROVED** [review]
code lgtm

File: src/main/webapp/app/exam/manage/exams/exam-update.component.ts (hunk 1 above)
Comment (SimonEntholzer) [reply src/main/webapp/app/exam/manage/exams/exam-update.component.ts:458]: Yeah, I also think this would be more polite

**Review by SimonEntholzer — APPROVED** [review]
lgtm

**Review by egekurt123 — APPROVED** [review]
Code LGTM

**Review by Strohgelaender — APPROVED** [review]

**Pull request merged**